ibis/bigquery/client.py

```diff
@@ -381,11 +363,13 @@ def exists_table(self, name, database=None):
 
     def list_tables(self, like=None, database=None):
         dataset = self._proxy.get_dataset(database or self.dataset_id)
```

Comment:
You don't actually need to make a GET call to fetch a Dataset resource. All you need to list is a reference, so you could do `dataset = client.dataset(database or self.dataset_id)`

Reply:
Great, I'll make this change.